tensorhive/controllers/reservation_event/CreateReservationEventController.py

```diff
@@ -0,0 +1,41 @@
+from tensorhive.models.reservation_event.ReservationEventModel import ReservationEventModel
+from tensorhive.models.user.UserModel import UserModel
+from connexion import NoContent
+from flask import jsonify
+from datetime import datetime
+
+
+class CreateReservationEventController():
+
+    @staticmethod
+    def create(reservation_event):
+        def parsed_datetime(input_datetime: str) -> str:
+            try:
+                return datetime.strptime(input_datetime, '%Y-%m-%dT%H:%M:%S.%fZ')
+            except ValueError:
+                return datetime.strptime(input_datetime, '%Y-%m-%dT%H:%M:%S')
+        if not UserModel.find_by_id(reservation_event['userId']):
+            return NoContent, 500
+
+        startTime = parsed_datetime(reservation_event['start'])
+        endTime = parsed_datetime(reservation_event['end'])
+        nodeId = reservation_event['nodeId']
+        userId = reservation_event['userId']
+        if (not UserModel.find_by_id(userId)):
+            return NoContent, 500
+
+        if (ReservationEventModel.find_events_the_same_time_nodes(startTime, endTime, nodeId) is not None):
```

Comment:
Maybe rename to:
```
find_node_events_between(start_time, end_time, node_id)???
find_node_colliding_events
find_events_with_time_collision
```

Reply:
I agree, but the same question as in api_specification: what is node, should it be a device?